src/daft-plan/src/planner.rs

```diff
@@ -80,6 +80,13 @@ pub fn plan(logical_plan: &LogicalPlan, cfg: Arc<DaftExecutionConfig>) -> DaftRe
             })) => {
                 let scan_tasks = scan_op.0.to_scan_tasks(pushdowns.clone())?;
 
+                let scan_tasks = daft_scan::scan_task_iters::split_by_row_groups(
+                    scan_tasks,
+                    cfg.parquet_split_row_groups_max_files,
+                    cfg.parquet_split_row_groups_threshold_bytes,
+                    cfg.parquet_split_row_groups_min_size_bytes,
+                );
```

Comment:
When this interacts with the below `merge_by_sizes`, it seems like split file scan tasks could get re-merged if there's no coordination between between task splitting and merging. E.g. if `parquet_split_row_groups_threshold_bytes < merge_scan_tasks_max_size_bytes`, as it is now, then split tasks may get re-merged, which can be both surprising to the splitting logic (undoing potential optimizations) and can lead to more inefficient reads.

Right now, if we had a scan task on a 512 MiB file with 8 64 MiB row groups, this `split_by_row_groups()` call would split this scan task into 8 scan tasks, and the below `merge_by_sizes()` would coalesce the scan tasks back into a single scan task, but instead has 8 separate file sources (each on a single row group) which will probably lead to a less efficient read.

Two options immediately come to mind for preventing this redundant splitting + merging:
1. Make the splitting threshold and merging max size bytes the same.
2. Modify the [`should_merge` determination in the merging logic](https://github.com/Eventual-Inc/Daft/blob/6cda37ae7032af8bc0342dc4291cf60be799a69f/src/daft-scan/src/scan_task_iters.rs#L60-L72) to not sources with defined chunk specs.

Lmk what you think!

Reply:
These are good questions. From what I can tell these values can reasonably be the same, but I don't think I know enough about the downstream effects of these thresholds to have a confident answer. @samster25 do you have any input on this?